test/module/shared_model/validators/field_validator_test.cpp

```diff
@@ -146,8 +156,13 @@ class FieldValidatorTest : public ValidatorsTest {
       if (!testcase.value_is_valid) {
         ASSERT_TRUE(!reason.second.empty())
             << testFailMessage(field_name, testcase.name);
-        EXPECT_EQ(testcase.expected_message, reason.second.at(0))
-            << testFailMessage(field_name, testcase.name);
+        if (not testcase.regexp)
+          EXPECT_EQ(testcase.expected_message, reason.second.at(0))
+              << testFailMessage(field_name, testcase.name);
+        else
+          ASSERT_THAT(reason.second.at(0),
```

Comment:
Is is possible to apply regex matching to all messages, so that `regexp` variable and additional branch are not required?